standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -851,8 +851,6 @@ private void updateTableColStats(RawStore rawStore, String catName, String dbNam
             sharedCache.refreshTableColStatsInCache(StringUtils.normalizeIdentifier(catName),
                 StringUtils.normalizeIdentifier(dbName), StringUtils.normalizeIdentifier(tblName),
                 tableColStats.getStatsObj());
-            // Update the table to get consistent stats state.
-            sharedCache.alterTableInCache(catName, dbName, tblName, table);
```

Comment:
Sorry, I am bit confused looking at this diff. So, the original issue seems to be - "Metastore's update service wrongly strips partition column stats from the cache in an attempt to update them." . How are we fixing this issue by not updating the stats in the cache ? Wouldn't the right fix for this is to to ensure sharedCache.alterTableInCache and sharedCache.alterPartitionInCache methods do the right thing and not incorrectly remove partition column stats ?

Reply:
If my understanding is correct the cache is already updated by the time we arrive here. Check my comment regarding `alterPartitionsInCache` and if something is not clear I can elaborate more.